api4/cloud_test.go

```diff
@@ -111,6 +111,90 @@ func Test_getCloudLimits(t *testing.T) {
 	})
 }
 
+func Test_GetSubscription(t *testing.T) {
+	subscription := &model.Subscription{
+		ID:          "MySubscriptionID",
+		CustomerID:  "MyCustomer",
+		ProductID:   "SomeProductId",
+		AddOns:      []string{},
+		StartAt:     1000000000,
+		EndAt:       2000000000,
+		CreateAt:    1000000000,
+		Seats:       10,
+		IsFreeTrial: "true",
+		DNS:         "some.dns.server",
+		IsPaidTier:  "false",
+		TrialEndAt:  2000000000,
+		LastInvoice: &model.Invoice{},
+	}
+
+	userFacingSubscription := &model.Subscription{
+		ID:          "MySubscriptionID",
+		CustomerID:  "",
+		ProductID:   "SomeProductId",
+		AddOns:      []string{},
+		StartAt:     0,
+		EndAt:       0,
+		CreateAt:    0,
+		Seats:       0,
+		IsFreeTrial: "true",
+		DNS:         "",
+		IsPaidTier:  "",
+		TrialEndAt:  2000000000,
+		LastInvoice: &model.Invoice{},
+	}
+
+	t.Run("NON Admin users receive the user facing subscription", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+
+		th.Client.Login(th.BasicUser.Email, th.BasicUser.Password)
+
+		th.App.Srv().SetLicense(model.NewTestLicense("cloud"))
+
+		cloud := mocks.CloudInterface{}
+
+		cloud.Mock.On("GetSubscription", mock.Anything).Return(userFacingSubscription, nil)
```

Comment:
This test doesn't achieve the intended purpose. The sanitizing happens within the handler and not from what the CWS returns

Reply:
you are right! nice catch @AGMETEOR